### PR TITLE
RUN-5469 Added state methods

### PR DESCRIPTION
**Pull request by pbaize**
#### Description of Change
Added methods to query BrowserView State
Tests https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5d7ff6957bacba7f712054bb
#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [x] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)
- [x] PR release notes describe the change in a way relevant to app-developers
- [x] PR has assigned reviewers

#### Release Notes

Notes: n/a experimental

### CODE REVIEW

File: src/api/application/application.ts [hunk 1]
```diff
@@ -440,7 +441,16 @@ export class Application extends EmitterBase<ApplicationEvents> {
...
             .then(({ payload }) => payload.data);
     }
-
+    /**
+    * Retrieves current application's views.
+    * @experimental
+    * @return {Promise.Array.<BrowserView>}
+    * @tutorial Application.getViews
+    */
+    public async getViews(): Promise<Array<BrowserView>> {
```
Comment (tomer-openfin) [inline src/api/application/application.ts]: not sure it's a good idea, but I'll say it - due to the impermanent nature of Views' target, wouldn't getCurrentViews be a better name choice?

File: src/api/window/window.ts [hunk 2]
```diff
@@ -737,6 +738,16 @@ export class _Window extends WebContents<WindowEvents> {
         return this.wire.sendAction('get-window-native-id', this.identity)
             .then(({ payload }) => payload.data);
     }
+    /**
+    * Retrieves window's attached views.
+    * @experimental
+    * @return {Promise.Array.<BrowserView>}
+    * @tutorial Window.getViews
+    */
+    public async getViews(): Promise<Array<BrowserView>> {
```
Comment (tomer-openfin) [inline src/api/window/window.ts]: Same as app.getViews comment

File: src/api/browserview/browserview.ts [hunk 3]
```diff
@@ -71,4 +72,15 @@ export class BrowserView extends WebContents<BaseEventMap> {
...
\ No newline at end of file
+
+    /**
+    * Retrieves the window the view is currently attached to.
+    * @experimental
+    * @return {Promise.Array.<_Window>}
+    * @tutorial BrowserView.getCurrentWindow
+    */
+    public getCurrentWindow = async () => {
+        const ack = await this.wire.sendAction('get-view-window', {...this.identity});
```
Comment (tomer-openfin) [inline src/api/browserview/browserview.ts]: Unlike the window-get-views action, this action is not typed

File: src/api/application/application.ts (hunk 1 above)
Comment (tomer-openfin) [reply src/api/application/application.ts]: oh right

**Comment by luiemilio** [comment]
LGTM

File: src/api/browserview/browserview.ts [hunk 4]
```diff
@@ -71,4 +72,15 @@ export class BrowserView extends WebContents<BaseEventMap> {
...
+
+    /**
+    * Retrieves the window the view is currently attached to.
+    * @experimental
+    * @return {Promise.<_Window>}
+    * @tutorial BrowserView.getCurrentWindow
+    */
+    public getCurrentWindow = async () => {
+        const ack = await this.wire.sendAction<{data: Identity}>('get-view-window', {...this.identity});
+        return new _Window(this.wire, ack.payload.data);
```
Comment (licui3936) [inline src/api/browserview/browserview.ts]: maybe destructuring:  const ack => const { payload }

**Pull request merged**